### PR TITLE
Move core/profiling/timers_preciseclock.h to core/support/ such that we

### PR DESCRIPTION
**Pull request by maxwell-demon**
have a gpr public function which returns cycle clock.

### CODE REVIEW

**Comment by ctiller** [comment]
Looks like include paths in basic_timers.c need to be updated.

**Comment by maxwell-demon** [comment]
You are right. Done.

**Comment by ctiller** [comment]
And it looks like this needs a merge with head.

**Comment by maxwell-demon** [comment]
done.

File: tools/run_tests/sources_and_headers.json [hunk 1]
```diff
@@ -12809,9 +12809,12 @@
       "src/core/json/json_reader.h", 
       "src/core/json/json_writer.h", 
       "src/core/profiling/timers.h", 
+<<<<<<< HEAD
```
Comment (ctiller) [inline tools/run_tests/sources_and_headers.json]: Merge conflict markers in code

File: vsprojects/grpc/grpc.vcxproj [hunk 2]
```diff
@@ -307,9 +307,12 @@
     <ClInclude Include="..\..\src\core\json\json_reader.h" />
     <ClInclude Include="..\..\src\core\json\json_writer.h" />
     <ClInclude Include="..\..\src\core\profiling\timers.h" />
+<<<<<<< HEAD
```
Comment (ctiller) [inline vsprojects/grpc/grpc.vcxproj]: Merge conflict

File: vsprojects/grpc/grpc.vcxproj.filters [hunk 3]
```diff
@@ -680,6 +680,8 @@
     <ClInclude Include="..\..\src\core\profiling\timers.h">
       <Filter>src\core\profiling</Filter>
     </ClInclude>
+<<<<<<< HEAD
```
Comment (ctiller) [inline vsprojects/grpc/grpc.vcxproj.filters]: Merge conflict

File: vsprojects/grpc_unsecure/grpc_unsecure.vcxproj [hunk 4]
```diff
@@ -290,9 +290,12 @@
     <ClInclude Include="..\..\src\core\json\json_reader.h" />
     <ClInclude Include="..\..\src\core\json\json_writer.h" />
     <ClInclude Include="..\..\src\core\profiling\timers.h" />
+<<<<<<< HEAD
```
Comment (ctiller) [inline vsprojects/grpc_unsecure/grpc_unsecure.vcxproj]: Merge conflict

File: vsprojects/grpc_unsecure/grpc_unsecure.vcxproj.filters [hunk 5]
```diff
@@ -578,6 +578,8 @@
     <ClInclude Include="..\..\src\core\profiling\timers.h">
       <Filter>src\core\profiling</Filter>
     </ClInclude>
+<<<<<<< HEAD
```
Comment (ctiller) [inline vsprojects/grpc_unsecure/grpc_unsecure.vcxproj.filters]: Merge conflict

**Comment by maxwell-demon** [comment]
Fixed.
Really sorry for the inconveniences...

**Pull request merged**